documentation/adr/0007-developer-scripts-storage-location.md

```diff
@@ -0,0 +1,53 @@
+[<-previous](0006-store-built-documentation-in-branch.md) | next->
+
+# 7. Developer scripts storage location
+
+Date: 2020-Jul-15
+
+## Status
+
+Proposed
+
+
+## Context
+
+When developing new algorithms and features for PACE-related software,
+developers often create useful demo/visualisation scripts for their own use.
+These scripts could be useful or interesting for other developers, and are
+important for reproducibility or justifying design decisions. They should be
+stored somewhere in version control so that they can be easily accessed by any
+developers and referred to later. However, they are not intended for general
+use and will not be actively maintained or tested. There are 2 main options:
+
+* Store them in a `dev_scripts` directory in each separate project repository
+* Store them in a `scripts` directory in `pace-developers`
+
+If they're in the `dev_scripts` directory for each project repository:
++ All in one place
++ Scripts will be close to the code they are used for
+- Scripts may not work with the version of the code they are distributed with
+- It's unclear where scripts that use more than one project would go
+- Despite the folder being called `dev_scripts` people might expect the scripts
+to actually work as they're in the main project repository
+
+If they're in a `scripts` directory in `pace-developers`:
++ They can be kept close to the decision-making developer documentation that
+they support
++ A version can be specified for any project dependencies
+
+## Decision
+
+Developer scripts will be stored in an appropriately placed `scripts`
+directory in the `pace-developers` repository. Depending on whether the
+script is tied to a particular software, or general algorithm development
+it could be stored in `pace-developers/euphonic/scripts` or
+`pace-developers/powder_averaging/scripts` for example.
+
+
+## Consequences
+
+* Scripts that are intended for development only should not be committed to
+the master branch of main software repositories
+* Any development scripts committed to `pace-developers` should specify commit
```

Comment:
The commit hash is less relevant than API version (e.g. the m.n release number for the project). 

Including the hash adds some additional information, particularly where these scripts are in active development against rapidly developing, unreleased APIs, but for legacy utilisation the version number is more transparent:
- a developer can quickly see a script written against the v1.3 API is unlikely to work without effort in the current master v3.4.n code
- the project documentation describing the API is versioned with the `m.n` project releases
- a git commit hash (e.g. `4e0510a`) gives the developer no information about compatibility or age and would require a search back through git history to find that information

Reply:
That's a fair point. We could have both, similar to the way [versioneer](https://github.com/warner/python-versioneer) does it e.g. `v0.11+g1076c97` which specifies both the most recent released version and the commit? (Versioneer actually also includes the number of commits since the last release too, but for projects that don't use versioneer that's not so easy to figure out, and we probably don't need that level of information).

Reply:
Seems that the hash is more useful immediately and the semver is more useful in the long run. I'd be happy to use the combined notation if people don't find it too cumbersome?